lib/diaspora/shareable.rb

```diff
@@ -57,7 +65,7 @@ def owned_or_visible_by_user(user)
         with_visibility.where(
           visible_by_user(user).or(arel_table[:public].eq(true)
                                      .or(arel_table[:author_id].eq(user.person_id)))
-        ).select("DISTINCT #{table_name}.*")
+        ).distinct_wrapper
```

Comment:
TBH I haven't checked whether this is needed for anything in the production code. You are right, it doesn't worth changing anything for tests only. I'll verify that then.